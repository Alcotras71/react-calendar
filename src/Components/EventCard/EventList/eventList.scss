$main-color: rgb(0, 166, 153);
$secondary-color: rgb(102, 226, 218);

.event-list {
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  height: 100%;
  font-size: 17px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;

  &__event-wrapper {
    overflow-y: auto;
    height: 250px;
  }

  &__event {
    margin-bottom: 10px;
  }

  &__title {
    font-size: 20px;
    font-weight: 600;
    line-height: 28px;
  }

  &__btn {
    width: 200px;
    height: 45px;
    margin: 0 auto;
    display: block;
    border-radius: 10px;
    background-color: $main-color;
    color: #fff;
    transition: all 0.5s ease;
    font-size: 20px;

    &:hover {
      color: #000;
      background-color: $secondary-color;
    }
  }
}

::-webkit-scrollbar {
  width: 12px;
}

::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
  border-radius: 10px;
}

::-webkit-scrollbar-thumb {
  border-radius: 10px;
  -webkit-box-shadow: inset 0 0 3px rgba(0, 0, 0, 0.5);
  background-color: $main-color;
}
