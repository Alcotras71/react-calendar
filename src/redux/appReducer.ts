const SHOW_LOADER = "app/SHOW_LOADER";
const HIDE_LOADER = "app/HIDE_LOADER";

//--------------------
type InitialState = {
  loading: boolean;
};
const initialState: InitialState = {
  loading: false,
};
//--------------------

//--------------------
export const appReducer = (
  state = initialState,
  action: ActionsTypes
): InitialState => {
  switch (action.type) {
    case SHOW_LOADER:
      return { ...state, loading: true };

    case HIDE_LOADER:
      return { ...state, loading: false };

    default:
      return state;
  }
};
//--------------------
type ActionsTypes = ShowLoader | HideLoader;
//--------------------

//--------------------
type ShowLoader = {
  type: typeof SHOW_LOADER;
};
export const showLoader = (): ShowLoader => {
  return {
    type: SHOW_LOADER,
  };
};
//--------------------
type HideLoader = {
  type: typeof HIDE_LOADER;
};
export const hideLoader = (): HideLoader => {
  return {
    type: HIDE_LOADER,
  };
};
