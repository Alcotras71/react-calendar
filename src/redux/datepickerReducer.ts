import React from "react";
import { EventData } from "../types/types";
import { returnFilteredDates } from "../utils/helpers/date-helpers";
import { updateObjectInArray } from "../utils/helpers/object-helpers";

const GET_DATE = "datepicker/GET_DATE";
const GET_EVENT_DATA = "datepicker/GET_EVENT_DATA";
const TOGGLE_CARD = "datepicker/TOGGLE_CARD";
const REMOVE_RECORD = "datepicker/REMOVE_RECORD";

export type CardType = "eventForm" | "eventList";

export const initialState = {
  touchedDate: null as Date | null,
  isOpen: false as boolean,
  eventData: [] as Array<EventData>,
  // localStorage.getItem("eventData")
  //   ? JSON.parse(localStorage.getItem("eventData") as string)
  //   : ([] as Array<EventData>),
  cardType: "eventList" as CardType,
};

export type DatepickerStateType = typeof initialState;

export const datepickerReducer = (
  state = initialState,
  action: ActionsTypes
): DatepickerStateType => {
  switch (action.type) {
    case GET_DATE:
      return {
        ...state,
        touchedDate: action.payload,
      };

    case TOGGLE_CARD:
      return {
        ...state,
        isOpen: action.payload.isOpen,
        cardType: action.payload.cardType,
      };

    case GET_EVENT_DATA:
      return {
        ...state,
        eventData: updateObjectInArray(state.eventData, action.payload),
      };

    case REMOVE_RECORD:
      return {
        ...state,
        eventData: returnFilteredDates(state.eventData, action.payload),
      };

    default:
      return state;
  }
};

type ActionsTypes = GetDate | ToggleCard | GetEventData | RemoveRecord;
//--------------------

//--------------------
type GetDate = {
  type: typeof GET_DATE;
  payload: Date;
};
export const getDate = (payload: Date): GetDate => ({
  type: GET_DATE,
  payload,
});
//--------------------
export type ToggleCardPayload = {
  isOpen: boolean;
  cardType: CardType;
};
type ToggleCard = {
  type: typeof TOGGLE_CARD;
  payload: ToggleCardPayload;
};
export const toggleCard = (payload: ToggleCardPayload): ToggleCard => ({
  type: TOGGLE_CARD,
  payload,
});
//--------------------
type GetEventData = {
  type: typeof GET_EVENT_DATA;
  payload: EventData;
};
export const getEventData = (payload: EventData): GetEventData => ({
  type: GET_EVENT_DATA,
  payload,
});
//--------------------
type RemoveRecord = {
  type: typeof REMOVE_RECORD;
  payload: EventData;
};
export const removeRecord = (payload: EventData): RemoveRecord => ({
  type: REMOVE_RECORD,
  payload,
});

export default datepickerReducer;
