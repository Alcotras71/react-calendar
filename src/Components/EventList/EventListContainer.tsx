import React, { ComponentType } from "react";
import { connect } from "react-redux";
import { AppStateType } from "../../redux/store";
import {
  toggleCard,
  getDate,
  getEventData,
  CardType,
} from "../../redux/datepickerReducer";
import EventList from "./EventList";
import { compose } from "redux";
import WithCardTemplate from "../../hoc/WithCardTemplate";
import { EventData } from "../../types/types";

type MapStatePropsType = {
  eventData: Array<EventData>;
  cardType: CardType;
};

type MapDispatchPropsType = {
  toggleCard: (payload: boolean) => void;
  getDate: (payload: Date) => void;
  getEventData: (payload: EventData) => void;
};

export type PropsType = MapDispatchPropsType & MapStatePropsType;

const mapStateToProps = (state: AppStateType): MapStatePropsType => {
  return {
    eventData: state.datepicker.eventData,
    cardType: state.datepicker.cardType,
  };
};

export default compose<ComponentType>(
  connect(mapStateToProps, {
    toggleCard,
    getDate,
    getEventData,
  }),
  WithCardTemplate
)(EventList);
