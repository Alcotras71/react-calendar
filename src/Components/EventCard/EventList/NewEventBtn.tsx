import React, { FC } from "react";
import { CardType, ToggleCardPayload } from "../../../redux/datepickerReducer";

type HandleToggleCardType = {
  isOpen: boolean;
  cardType: CardType;
};

type NewEventBtn = {
  isOpen: boolean;
  cardType: CardType;
  toggleCard: (payload: ToggleCardPayload) => void;
};

const NewEventBtn: FC<NewEventBtn> = ({
  isOpen,
  cardType,
  toggleCard,
  children,
}) => {
  const handleToggleCard = ({ isOpen, cardType }: HandleToggleCardType) => {
    toggleCard({ isOpen, cardType });
  };

  return (
    <button
      onClick={() => handleToggleCard({ isOpen, cardType })}
      className="event-list__btn"
      type="submit"
    >
      {children}
    </button>
  );
};

export default NewEventBtn;
