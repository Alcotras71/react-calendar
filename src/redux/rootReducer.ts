import { combineReducers } from "redux";
import { appReducer } from "./appReducer";
import { datepickerReducer } from "./datepickerReducer";
import { weatherReducer } from "./weatherReducer";

export const rootReducer = combineReducers({
  app: appReducer,
  datepicker: datepickerReducer,
  weather: weatherReducer,
});

type RootReducerType = typeof rootReducer;
export type AppStateType = ReturnType<RootReducerType>;
