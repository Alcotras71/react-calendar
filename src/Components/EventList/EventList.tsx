import React, { FC, SyntheticEvent, useState } from "react";
import DatePicker, { registerLocale } from "react-datepicker";
import TimePicker, { TimePickerValue } from "react-time-picker";
import { PropsType } from "./EventListContainer";
import ru from "date-fns/locale/ru";


const EventList: FC<PropsType> = ({ cardType }) => {
  return cardType === "eventList" ? (
    <button className="user-card__btn" type="submit">
      Создать
    </button>
  ) : null;
};

export default EventList;
