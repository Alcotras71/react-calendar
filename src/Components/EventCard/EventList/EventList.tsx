import moment from "moment";
import React, { FC } from "react";
import { CardType, ToggleCardPayload } from "../../../redux/datepickerReducer";
import { EventData } from "../../../types/types";
import NewEventBtn from "./NewEventBtn";

import "./eventList.scss";

export type PropsType = {
  pickedData: Array<EventData> | undefined;
  toggleCard: (payload: ToggleCardPayload) => void;
  removeRecord: (payload: EventData) => void;
};

const EventList: FC<PropsType> = ({ pickedData, toggleCard, removeRecord }) => {
  const handleRemoveRecord = (pickedData: Array<EventData>, index: number) => {
    removeRecord(pickedData && pickedData[index]);
  };

  return (
    <div className="event-list">
      <div className="event-list__event-wrapper">
        {pickedData?.map((data, index) => {
          return (
            <div className="event-list__event" key={index}>
              <div className="event-list__container">
                <h2 className="event-list__title">{data?.eventName}</h2>
                <button
                  className={`event-list__btn-delete event-list__btn-${index}`}
                  onClick={() => handleRemoveRecord(pickedData, index)}
                >
                  X
                </button>
              </div>
              <div className="event-list__clock">
                <span>{data?.startTime}</span> - <span>{data?.endTime}</span>
              </div>
              <p className="event-list__date">
                {moment(data?.touchedDate).format("DD-MM-yyyy")}
              </p>
            </div>
          );
        })}
      </div>

      <NewEventBtn isOpen={true} cardType={"eventForm"} toggleCard={toggleCard}>
        Новое событие
      </NewEventBtn>
    </div>
  );
};

export default EventList;
