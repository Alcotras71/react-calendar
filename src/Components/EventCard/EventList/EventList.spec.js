import React from "react";
import EventList from "./EventList";

import { toggleCard, removeRecord } from "../../../redux/datepickerReducer";
import { mockEventData } from "../../../../__mocks__/mock";

describe("EventList component", () => {
  const setUp = (props) => shallow(<EventList {...props} />);

  describe("render EventList", () => {
    it("render EventList without props", () => {
      const component = setUp();
      expect(component).toMatchSnapshot();
    });

    it("render EventList with props", () => {
      const component = setUp({
        pickedData: mockEventData.eventDataArr,
        toggleCard,
        removeRecord,
      });
      expect(component).toMatchSnapshot();
    });
  });

  describe('check EventList functions', () => {
    
  })
});
