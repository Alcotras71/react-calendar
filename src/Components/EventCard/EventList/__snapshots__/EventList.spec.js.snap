// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList component render EventList render EventList with props 1`] = `
<div
  className="event-list"
>
  <div
    className="event-list__event-wrapper"
  >
    <div
      className="event-list__event"
      key="0"
    >
      <div
        className="event-list__container"
      >
        <h2
          className="event-list__title"
        >
          mocki
        </h2>
        <button
          className="event-list__btn-delete event-list__btn-0"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="event-list__clock"
      >
        <span>
          12:00
        </span>
         - 
        <span>
          12:30
        </span>
      </div>
      <p
        className="event-list__date"
      >
        28-07-2021
      </p>
    </div>
    <div
      className="event-list__event"
      key="1"
    >
      <div
        className="event-list__container"
      >
        <h2
          className="event-list__title"
        >
          Loki-moki
        </h2>
        <button
          className="event-list__btn-delete event-list__btn-1"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="event-list__clock"
      >
        <span>
          16:00
        </span>
         - 
        <span>
          18:30
        </span>
      </div>
      <p
        className="event-list__date"
      >
        28-09-2021
      </p>
    </div>
    <div
      className="event-list__event"
      key="2"
    >
      <div
        className="event-list__container"
      >
        <h2
          className="event-list__title"
        >
          PockMocki
        </h2>
        <button
          className="event-list__btn-delete event-list__btn-2"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="event-list__clock"
      >
        <span>
          9:00
        </span>
         - 
        <span>
          10:30
        </span>
      </div>
      <p
        className="event-list__date"
      >
        30-07-2021
      </p>
    </div>
  </div>
  <button
    className="event-list__btn"
    onClick={[Function]}
    type="submit"
  >
    Новое событие
  </button>
</div>
`;

exports[`EventList component render EventList render EventList without props 1`] = `
<div
  className="event-list"
>
  <div
    className="event-list__event-wrapper"
  />
  <button
    className="event-list__btn"
    onClick={[Function]}
    type="submit"
  >
    Новое событие
  </button>
</div>
`;
